parsers/v.c

```diff
@@ -0,0 +1,2892 @@
+}
+
+// alias: 'type' type ['[' type ']']? '=' vtype ['|' vtype]*
+static void parseAlias (tokenInfo *const token, vString *const access, int scope)
+{
+	Assert (isKeyword (token, KEYWORD_type));
+	PARSER_PROLOGUE ("alias");
+
+	readToken (token);
+	if ((!IsBuiltin && expectToken (token, TOKEN_TYPE)) ||
+	    (IsBuiltin && (expectToken (token, TOKEN_TYPE, TOKEN_KEYWORD) ||
+	                   expectKeyword (token, KEYWORD_TYPE))))
+	{
+		int newScope = makeTagEx (token, NULL, KIND_ALIAS, scope, access);
+		registerEntry (newScope);
+		readToken (token);
+		if (isToken (token, TOKEN_OPEN_SQUARE)) // template args
+		{
+			skipToToken (TOKEN_CLOSE_SQUARE, NULL);
+			readToken (token);
+		}
+		if (expectToken (token, TOKEN_ASSIGN))
+		{
+			readToken (token);
+			if(parseVType (token, NULL, scope, false, false))
+				readToken (token);
+			while (true)
+			{
+				if (isToken (token, TOKEN_PIPE))
+				{
+					readToken (token);
+					if (parseVType (token, NULL, scope, false, false))
+					{
+						readToken (token);
+						continue;
+					}
+					else
+						unreadToken (token);
+				}
+				else
+					unreadToken (token);
+				break;
+			}
+		}
+	}
+
+	PARSER_EPILOGUE ();
+}
+
+// list: [
+//     [access* indent [',' access* ident]* [':=' | 'in'] list] |
+//     [expr? [',' expr]*]
+// ]
+static void parseExprList (tokenInfo *const token, int scope,
+                           vString *const access, bool hasInDecl)
+{
+	PARSER_PROLOGUE ("list");
+
+	// attempt to consume multi-var declaration (special case of list)
+	bool consumedIdent = true;
+	if (isToken (token, TOKEN_IDENT))
+	{
+		tokenInfo *identToken = dupToken (token);
+		vString *accesses = accesses = access? access : vStringNew ();
+		vString *idents = token->string;
+		token->string = NULL;
+		readToken (token);
+		if (isToken (token, TOKEN_COMMA))
+		{
+			do
+			{
+				vStringPut (accesses, ',');
+				vStringPut (idents, ',');
+				readToken (token);
+				skipAccessAndReadToken (token, &accesses);
+				if (!isToken (token, TOKEN_IDENT))
+					consumedIdent = false;
+				else
+				{
+					deleteToken (identToken);
+					identToken = dupToken (token);
+					vStringCat (idents, token->string);
+					readToken (token);
+				}
+			}
+			while (consumedIdent && isToken (token, TOKEN_COMMA));
+		}
+		if (consumedIdent)
+		{
+			if (isToken (token, TOKEN_DECLARE) ||
+			    (hasInDecl && isKeyword (token, KEYWORD_in)))
+				parseDeclare (token, idents, scope, accesses);
+			else
+			{
+				if (isToken (token, TOKEN_DOT))
+				{
+					unreadToken (token);
+					parseFullyQualified (identToken, false);
+					if (isToken (identToken, TOKEN_TYPE))
+						parseVType (identToken, NULL, scope, false, true);
+					readToken (token);
+				}
+				if (!parseExprCont (token, scope, true))
+					unreadToken (token);
+			}
+		}
+		deleteToken (identToken);
+		vStringDelete (idents);
+		if (!access)
+			vStringDelete (accesses);
+	}
+	else
+		parseExpression (token, scope, NULL);
+
+    while (true)
```

Comment:
Strange depth of indentation.

![image](https://github.com/universal-ctags/ctags/assets/77077/11fa7efc-2973-4762-9f89-dcb6b9e67c12)

Reply:
Yeah, `whitespace-mode` shows something interesting. If I hit TAB at the start of this line:
<img width="536" alt="Screenshot 2023-08-15 at 11 22 34 pm" src="https://github.com/universal-ctags/ctags/assets/3274122/e5f52f25-eda4-4c78-b5ad-6fea0a897c31">
Emacs converts the tabs before `TOKEN_CLOSE_CURLY` to spaces! I presume this is to stop the indentation depending on tab-width. I've converted all spaces to tabs now though anyway.